charts/vela-core/templates/test/test-application.yaml

```diff
@@ -0,0 +1,62 @@
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  annotations:
+    helm.sh/hook: test-success
+  name: first-vela-app
+spec:
+  components:
+    - name: express-server
+      type: webservice
+      properties:
+        image: crccheck/hello-world
+        port: 8000
+      traits:
+        - type: ingress
+          properties:
+            domain: testsvc.example.com
+            http:
+              "/": 8000
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ .Release.Name }}-application-test"
+  annotations:
+    "helm.sh/hook": test
+spec:
+  serviceAccountName: kubevela-vela-core
+  containers:
+    - name: {{ .Release.Name }}-application-test
+      image: alpine/k8s:1.18.2
+      imagePullPolicy: IfNotPresent
+      command:
+        - /bin/bash
+        - -ec
+        - |
+
+          set -e
+
+          echo "Waiting application is ready..."
+          # wait for controller ready
+          while ! [ `kubectl -n vela-system get application | grep -v NAME | wc -l` = 1 ]; do
+            echo "waiting for application installed"
+            sleep 1
+          done
+
+          # wait for deploy being created
+          while ! [ `kubectl -n vela-system get deploy express-server | grep -v NAME | wc -l` = 1 ]; do
+            echo "waiting for deployment being created"
+            sleep 1
+          done
+
+          # wait for ingress being created
+          while ! [ `kubectl -n vela-system get ing express-server | grep -v NAME | wc -l` = 1 ]; do
+            echo "waiting for ingress being created"
+            sleep 1
+          done
```

Comment:
will the `helm test` fail if the job is hanging on sleep? Do we have any timeout settings?

Reply:
Yeah the timeout is `5m` for now.

Reply:
That's good